// admin-bro/lib/frontend/components/app/records-table-header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _propertyHeader = _interopRequireDefault(require("./property-header"));

var _types = require("../../types");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const RecordsTableHeader = props => {
  const {
    titleProperty,
    properties,
    sortBy,
    direction
  } = props;
  return _react.default.createElement("thead", null, _react.default.createElement("tr", {
    key: "header"
  }, properties.map(property => _react.default.createElement(_propertyHeader.default, {
    key: property.name,
    titleProperty: titleProperty,
    property: property,
    sortBy: sortBy,
    direction: direction
  })), _react.default.createElement("th", {
    key: "actions",
    style: {
      width: 80
    }
  })));
};

RecordsTableHeader.propTypes = {
  /**
   * {@link BaseProperty~JSON}
   */
  titleProperty: _types.propertyType.isRequired,

  /**
   * Array of {@link BaseProperty~JSON}
   */
  properties: _propTypes.default.arrayOf(_types.propertyType).isRequired,

  /**
   * Name of the property which should be marked as currently sorted by
   */
  sortBy: _propTypes.default.string.isRequired,

  /**
   * Sort direction
   */
  direction: _propTypes.default.oneOf(['asc', 'desc']).isRequired
};
var _default = RecordsTableHeader;
exports.default = _default;
