// admin-bro/lib/frontend/components/property-type/reference/reference-value.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

var _reactRedux = require("react-redux");

var _viewHelpers = _interopRequireDefault(require("../../../../backend/utils/view-helpers"));

var _types = require("../../../types");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ReferenceValue = props => {
  const {
    resources,
    property,
    record
  } = props;
  const h = new _viewHelpers.default();
  const refId = record.params[property.name];
  const populated = record.populated[property.name];
  const value = populated && populated.title || refId;
  const referenceResource = resources.find(r => r.id === property.reference);

  if (referenceResource.recordActions.find(a => a.name === 'show') && populated) {
    const href = h.recordActionUrl({
      resourceId: property.reference,
      recordId: refId,
      actionName: 'show'
    });
    return _react.default.createElement(_reactRouterDom.Link, {
      to: href
    }, value);
  }

  return _react.default.createElement("span", null, value);
};

const mapStateToProps = state => ({
  resources: state.resources
});

ReferenceValue.propTypes = {
  property: _types.propertyType.isRequired,
  record: _types.recordType.isRequired,
  resources: _propTypes.default.arrayOf(_types.resourceType).isRequired
};

var _default = (0, _reactRedux.connect)(mapStateToProps)(ReferenceValue);

exports.default = _default;
